### PR TITLE
BinarySearchTree.py

### PR DESCRIPTION
**Pull request by avsingh999**
Fixes #{ISSUE_NUMBER}

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [ ] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by singhpratyush** [comment]
Hi! Thanks for showing interest in contributing to this repository.

Please make sure that you have ticked the points mentioned in PR description correctly.

Thanks again!

**Comment by avsingh999** [comment]
what is wrong in code??

**Comment by singhpratyush** [comment]
@avsingh999: There are some style issues with the code. You can see more details [here](https://travis-ci.org/iiitv/algos/builds/290349855).

File: binary_search_tree/BinarySearchTree.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
+class Node(object):
+# Node for Binary search Tree and set the Node
+    def __init__(self, data):
+        self.data = data
```
Comment (singhpratyush) [inline binary_search_tree/BinarySearchTree.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `pep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmphwomd5ps/binary_search_tree/BinarySearchTree.py
+++ b/tmp/tmphwomd5ps/binary_search_tree/BinarySearchTree.py
@@ -1,5 +1,5 @@
 class Node(object):
-# Node for Binary search Tree and set the Node
+    # Node for Binary search Tree and set the Node
     def __init__(self, data):
         self.data = data
         self.left = None
```

File: binary_search_tree/BinarySearchTree.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        # data node is found
+        else:
+            if root.left is None:
+                return root.right
+            elif root.right is None:
+                return root.left
+            # if left or right child is Not None find the the minimum node in right child
+            temp = Bst.minright(root.right)
+            root.data = temp.data
+            # recursive delete the minimum node in right child
```
Comment (singhpratyush) [inline binary_search_tree/BinarySearchTree.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `pep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmphwomd5ps/binary_search_tree/BinarySearchTree.py
+++ b/tmp/tmphwomd5ps/binary_search_tree/BinarySearchTree.py
@@ -46,9 +46,8 @@
             node = Bst.min_right(node.left)
         return node
 
-
     # delete a node
-    def delete(root,data):
+    def delete(root, data):
         if root is None:
             return root
         if data < root.data:
```

File: binary_search_tree/BinarySearchTree.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+    # print postorder
+    def postorder(root):
+        if root is not None:
+            Bst.preorder(root.left)
+            Bst.preorder(root.right)
+            print(root.data)
+
+
+# check the method
```
Comment (singhpratyush) [inline binary_search_tree/BinarySearchTree.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `pep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmphwomd5ps/binary_search_tree/BinarySearchTree.py
+++ b/tmp/tmphwomd5ps/binary_search_tree/BinarySearchTree.py
@@ -68,7 +68,6 @@
             root.right = Bst.delete(root.right, temp.data)
         return root
 
-
     # print inorder
     def inorder(root):
         if root is not None:
```

File: binary_search_tree/BinarySearchTree.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+def main():
+    root = None
+    root = Bst.insert(root, 5)
+    root = Bst.insert(root, 11)
+    root = Bst.insert(root, 3)
+    root = Bst.insert(root, 1)
+    root = Bst.insert(root, 50)
+    root = Bst.insert(root, 45)
+    root = Bst.insert(root, 30)
```
Comment (singhpratyush) [inline binary_search_tree/BinarySearchTree.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `pep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmphwomd5ps/binary_search_tree/BinarySearchTree.py
+++ b/tmp/tmphwomd5ps/binary_search_tree/BinarySearchTree.py
@@ -76,7 +76,6 @@
             print(root.data)
             Bst.inorder(root.right)
 
-
     # print preorder
     def preorder(root):
         if root is not None:
```

File: binary_search_tree/BinarySearchTree.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    root = Bst.insert(root, 35)
+    print("****** InOrder ******")
+    Bst.inorder(root)
+    print("****** PreOrder ******")
+    Bst.preorder(root)
+    print("****** PostOrder ******")
+    Bst.postorder(root)
+    print("***** search Node ******")
```
Comment (singhpratyush) [inline binary_search_tree/BinarySearchTree.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `pep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmphwomd5ps/binary_search_tree/BinarySearchTree.py
+++ b/tmp/tmphwomd5ps/binary_search_tree/BinarySearchTree.py
@@ -84,7 +84,6 @@
             Bst.preorder(root.left)
             Bst.preorder(root.right)
 
-
     # print postorder
     def postorder(root):
         if root is not None:
```

**Pull request closed without merging**